src/scripts/templates.js

```diff
@@ -1,238 +1,192 @@
-import { strToEl } from './lib/utils';
-
 export const TEMPLATES = {
   containerOuter(
-    globalClasses,
-    direction,
+    { containerOuter },
+    dir,
     isSelectElement,
     isSelectOneElement,
     searchEnabled,
     passedElementType,
   ) {
-    const tabIndex = isSelectOneElement ? 'tabindex="0"' : '';
-    let role = isSelectElement ? 'role="listbox"' : '';
-    let ariaAutoComplete = '';
-
-    if (isSelectElement && searchEnabled) {
-      role = 'role="combobox"';
-      ariaAutoComplete = 'aria-autocomplete="list"';
+    const div = Object.assign(document.createElement('div'), {
+      className: containerOuter,
+      dir,
+    });
+    div.dataset.type = passedElementType;
+    if (isSelectOneElement) div.tabIndex = 0;
+    if (isSelectElement) {
+      div.setAttribute('role', searchEnabled ? 'combobox' : 'listbox');
+      if (searchEnabled) div.setAttribute('aria-autocomplete', 'list');
     }
+    div.setAttribute('aria-haspopup', 'true');
+    div.setAttribute('aria-expanded', 'false');
 
-    return strToEl(`
-      <div
-        class="${globalClasses.containerOuter}"
-        data-type="${passedElementType}"
-        ${role}
-        ${tabIndex}
-        ${ariaAutoComplete}
-        aria-haspopup="true"
-        aria-expanded="false"
-        dir="${direction}"
-        >
-      </div>
-    `);
+    return div;
   },
-  containerInner(globalClasses) {
-    return strToEl(`
-      <div class="${globalClasses.containerInner}"></div>
-    `);
+  containerInner({ containerInner }) {
+    return Object.assign(document.createElement('div'), {
+      className: containerInner,
+    });
   },
-  itemList(globalClasses, isSelectOneElement) {
-    const localClasses = classNames(globalClasses.list, {
-      [globalClasses.listSingle]: isSelectOneElement,
-      [globalClasses.listItems]: !isSelectOneElement,
+  itemList({ list, listSingle, listItems }, isSelectOneElement) {
+    return Object.assign(document.createElement('div'), {
+      className: `${list} ${isSelectOneElement ? listSingle : listItems}`,
     });
-
-    return strToEl(`
-      <div class="${localClasses}"></div>
-    `);
   },
-  placeholder(globalClasses, value) {
-    return strToEl(`
-      <div class="${globalClasses.placeholder}">
-        ${value}
-      </div>
-    `);
+  placeholder({ placeholder }, value) {
+    return Object.assign(document.createElement('div'), {
+      className: placeholder,
+      innerHTML: value,
+    });
   },
-  item(globalClasses, data, removeItemButton) {
-    const ariaSelected = data.active ? 'aria-selected="true"' : '';
-    const ariaDisabled = data.disabled ? 'aria-disabled="true"' : '';
 
-    let localClasses = classNames(globalClasses.item, {
-      [globalClasses.highlightedState]: data.highlighted,
-      [globalClasses.itemSelectable]: !data.highlighted,
-      [globalClasses.placeholder]: data.placeholder,
+  item(
+    { item, button, highlightedState, itemSelectable, placeholder },
```

Comment:
Subjective thing but I think this almost makes the code less readable as it's not immediately obvious their class names

Reply:
Yes, in that particular case may be, but I'd keep it like this just for uniformity with the rest of callbacks here.